packages/simcore-sdk/src/simcore_sdk/node_ports_common/aws_s3_cli.py

```diff
@@ -0,0 +1,307 @@
+            returncode=proc.returncode,
+        )
+
+    _logger.debug("'%s' result:\n%s", str_cmd, command_output)
+    return command_output
+
+
+def _get_exclude_filters(exclude_patterns: set[str] | None) -> list[str]:
+    if exclude_patterns is None:
+        return []
+
+    exclude_options: list[str] = []
+    for entry in exclude_patterns:
+        exclude_options.append("--exclude")
+        exclude_options.append(entry.replace("*", "**"))
+
+    return exclude_options
+
+
+async def _get_s3_folder_size(
+    aws_s3_cli_settings: AwsS3CliSettings,
+    *,
+    s3_path: str,
+) -> ByteSize:
+    cli_command = [
+        "aws",
+        "s3",
+        "ls",
+        "--summarize",
+        "--recursive",
+        s3_path,
+        "| grep 'Total Size' | awk '{print $3}'",
+    ]
+
+    if aws_s3_cli_settings.AWS_S3_CLI_PROVIDER.value != S3Provider.AWS:
+        cli_command.insert(
+            1, f"--endpoint-url {aws_s3_cli_settings.AWS_S3_CLI_S3.S3_ENDPOINT}"
+        )
+
+    result = await _async_aws_cli_command(
+        *cli_command, aws_s3_cli_settings=aws_s3_cli_settings
+    )
+    return ByteSize(result.strip())
+
+
+def _get_size_and_manage_symlink(path: Path) -> ByteSize:
+    if path.is_symlink():
+        # Convert symlink to a .osparclink file that can be stored in the S3
+        target_path = os.readlink(path)
+        _name = path.name + f".{_CONVERT_SYMLINK_TO_OSPARC_LINK}"
+
+        textfile_path = path.parent / _name
+        with open(textfile_path, "w") as file:
+            file.write(target_path)
+        return ByteSize(0)
+    return ByteSize(path.stat().st_size)
+
+
+async def _get_local_folder_size(local_path: Path) -> ByteSize:
+    total_size = 0
+    for dirpath, _, filenames in os.walk(local_path):
+        for filename in filenames:
+            file_path = Path(dirpath) / filename
+            total_size += _get_size_and_manage_symlink(Path(file_path))
+    return ByteSize(total_size)
+
+
+async def _sync_sources(
+    aws_s3_cli_settings: AwsS3CliSettings,
+    progress_bar: ProgressBarData,
+    *,
+    source: str,
+    destination: str,
+    local_dir: Path,
+    exclude_patterns: set[str] | None,
+    debug_logs: bool = True,
+) -> None:
+
+    if source.startswith("s3://"):
+        folder_size: ByteSize = await _get_s3_folder_size(
+            aws_s3_cli_settings, s3_path=shlex.quote(source)
+        )
+    else:
+        folder_size = await _get_local_folder_size(Path(source))
+
+    cli_command = [
+        "aws",
+        "s3",
+        "sync",
+        "--delete",
+        shlex.quote(source),  # <-- source
+        shlex.quote(destination),  # <-- destination
+        # filter options
+        *_get_exclude_filters(exclude_patterns),
+        "--no-follow-symlinks",
+    ]
+
+    if not aws_s3_cli_settings.AWS_S3_CLI_PROVIDER.value == S3Provider.AWS:
+        cli_command.insert(
+            1, f"--endpoint-url {aws_s3_cli_settings.AWS_S3_CLI_S3.S3_ENDPOINT}"
```

Comment:
same here. in AWS deploys the S3_ENDPOINT is None. so you do not need the provider